### PR TITLE
Cherry pick for #635 form feedback

### PR DESCRIPTION
**Pull request by lenglund**

### CODE REVIEW

File: compair/static/modules/course/course-module.js [hunk 1]
```diff
@@ -704,11 +793,216 @@ module.controller(
...
+               
+                if (firstDate.date && secondDate.date && firstDate.time && secondDate.time) {
+                   
+                    // is the date the same?
+                    if (firstDate.date.toDateString() === secondDate.date.toDateString()) {
+
+                            // can the start and end time be the same?
+                            if (canBeEqual) {
+                                // does the end time follow or equal the start time?
+                                if (firstDate.time.toTimeString().split(' ')[0] <= secondDate.time.toTimeString().split(' ')[0]) {
```
[the remark this answers is not in the document]
Comment (lenglund) [reply compair/static/modules/course/course-module.js]: Adopted some changes applied for assignment module; hopefully that does the trick.

[the remark this answers is not in the document]
Comment (kitsook) [reply compair/static/modules/course/course-module.js]: Still encounter errors if the start date == end date.  Scenario:
- the original course has both start date and end date defined
- duplicate the course
- for the duplicated course, change the start date and end date to be the same. don't change the time
- click next and encounter javascript error:
```
Error: courseStart.time.toTimeString is not a function
$scope.showFirstErrors@http://localhost:8080/app/modules/course/course-module.js:858:109
......
```

However, if tried to modify both start and end time before clicking next, it works.  So seems that the start/end time variables are not cloned properly from the original course.

This block only copies the date part if `$scope.originalCourse.start_date` is set.  Tried to copy the start/end time too and the problem goes away.  Could be the cause.
https://github.com/ubc/compair/blob/15ae4e2e3f0cc901da526cc4a639bc686558a36c/compair/static/modules/course/course-module.js#L654-L665

**Comment by kitsook** [comment]
Still encounter some errors on a special scenario (duplicate course start == course end.  without changing the start/end time).  See [comments above](https://github.com/ubc/compair/pull/886#discussion_r308354313).

Otherwise LGTM.

File: compair/static/modules/course/course-module.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (lenglund) [reply compair/static/modules/course/course-module.js]: Thanks for investigating. I made a change here; can you test again?

**Comment by lenglund** [comment]
For some reason, getting internal server error when I use this branch now? Just my local?

**Comment by lenglund** [comment]
Ready for re-review!

**Comment by lenglund** [comment]
Sorry, just realized I forgot a form. Can you review the last commit? Thx!

**Pull request merged**